extensions/Worker.Extensions.Http.AspNetCore.Analyzers/HttpResultAttributeExpected.cs

```diff
@@ -0,0 +1,97 @@
+                .SelectMany(attrList => attrList.Attributes)
+                .Where(attr => SymbolEqualityComparer.Default.Equals(semanticModel.GetTypeInfo(attr).Type, functionAttributeSymbol));
+
+            if (!functionNameAttribute.Any())
+            {
+                return;
+            }
+
+            var functionName = functionNameAttribute.First().ArgumentList.Arguments[0]; // only one argument in FunctionAttribute which is the function name
+
+            var httpTriggerAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(HttpTriggerAttributeFullName);
+            var hasHttpTriggerAttribute = methodDeclaration.ParameterList.Parameters
+                .SelectMany(param => param.AttributeLists)
+                .SelectMany(attrList => attrList.Attributes)
+                .Select(attr => semanticModel.GetTypeInfo(attr).Type)
+                .Any(attrSymbol => SymbolEqualityComparer.Default.Equals(attrSymbol, httpTriggerAttributeSymbol));
+
+            if (!hasHttpTriggerAttribute)
+            {
+                return;
+            }
+
+            var returnType = methodDeclaration.ReturnType;
+            var returnTypeSymbol = semanticModel.GetTypeInfo(returnType).Type;
+
+            var taskSymbol = semanticModel.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1");
+            var isGenericTask = returnTypeSymbol != null && returnTypeSymbol.OriginalDefinition.Equals(taskSymbol, SymbolEqualityComparer.Default);
+            var isTaskOfCustomType = isGenericTask && returnTypeSymbol is INamedTypeSymbol genericSymbol
+                && IsCustomType(genericSymbol.TypeArguments[0]);
+            var isCustomReturnType = IsCustomType(returnTypeSymbol);
+
+            if (!(isTaskOfCustomType || isCustomReturnType))
+            {
+                return;
+            }
+
+            var httpResponseDataSymbol = semanticModel.Compilation.GetTypeByMetadataName(HttpResponseDataFullName);
+            var hasHttpResponseData = returnTypeSymbol.GetMembers()
+                .OfType<IPropertySymbol>()
+                .Any(prop => SymbolEqualityComparer.Default.Equals(prop.Type, httpResponseDataSymbol));
+
+            var httpResultAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(HttpResultAttributeFullName);
+            var hasHttpResultAttribute = returnTypeSymbol.GetMembers()
+                .SelectMany(member => member.GetAttributes())
+                .Any(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, httpResultAttributeSymbol));
+
+            if (!hasHttpResultAttribute && !hasHttpResponseData)
+            {
+                var diagnostic = Diagnostic.Create(DiagnosticDescriptors.MultipleOutputHttpTriggerWithoutHttpResultAttribute, methodDeclaration.ReturnType.GetLocation(), functionName.ToString());
+                context.ReportDiagnostic(diagnostic);
+            }
+        }
+
+        private static bool IsCustomType(ISymbol symbol)
```

Comment:
Would like reviews/ideas on how to approach determining whether or not a return type is a POCO.